File: easy/python/389.py
# https://leetcode.com/problems/find-the-difference/
# Time: O(n)
# Space: O(1)
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        s_map = {}
        for letter in s:
            if letter not in s_map:
                s_map[letter] = 1
            else:
                s_map[letter] += 1
        for letter in t:
            if letter not in s_map or s_map[letter] == 0:
                return letter
            else:
                s_map[letter] -= 1

File: easy/ts/389.ts
// https://leetcode.com/problems/find-the-difference/
// Time: O(n)
// Space: O(1)
function findTheDifference(s: string, t: string): string {
  let sMap = {};
  for (let i = 0; i < s.length; i++) {
    if (!sMap[s[i]]) {
      sMap[s[i]] = 1;
    } else {
      sMap[s[i]]++;
    }
  }
  for (let i = 0; i < t.length; i++) {
    let letter = t[i];
    if (!sMap[letter] || sMap[letter] === 0) return letter;
    sMap[letter]--;
  }
  return "";
}
